### PR TITLE
EDM-694: FlightCtl deployment with Quadlets

### PR DESCRIPTION
**Pull request by oglok**
Create Quadlet files for the basic FlightCtl services to be able to deploy in a local machine with a Makefile target.

There are two Makefile targets:

`make deploy-quadlets`
`make clean-quadlets`

### CODE REVIEW

File: deploy/deploy.mk [hunk 1]
```diff
@@ -32,6 +32,15 @@ deploy-mq:
 	podman rm -f flightctl-mq || true
 	cd deploy/podman && podman-compose up -d flightctl-mq
 
+deploy-quadlets:
```
Comment (hexfusion) [inline deploy/deploy.mk:35]: :+1:

**Review by fzdarsky — COMMENTED** [review]
Nice, this looks like a very clean approach and I love that we can let systemd figure out startup deps.

Could we remove the docker-compose path in `/deploy/podman` at the same time?

Then, I have second thoughts about putting it under `/deploy/quadlets` (wondering how many people know that name and whether the official name didn't change to "podman-systemd units" instead). So wanted to check what makes most sense: `quadlets`, `systemd`, `podman`?

File: deploy/quadlets/flightctl-api/flightctl-api.container [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+Network=flightctl.network
+Environment=HOME=/root
+Environment=FLIGHTCTL_DISABLE_AUTH=true
+
+PublishPort=3443:3443
+PublishPort=7443:7443
+PublishPort=7444:7444
+PublishPort=15690:15690
+PublishPort=15691:15691
+Volume=./flightctl-api-config/config.yaml:/root/.flightctl/config.yaml
```
[the remark this answers is not in the document]
Comment (mangelajo) [reply deploy/quadlets/flightctl-api/flightctl-api.container:20]: I think a volume unit would make sense if we want to store data, but for a config file? it's just a mapping

**Comment by oglok** [comment]
> Nice, this looks like a very clean approach and I love that we can let systemd figure out startup deps.
> 
> Could we remove the docker-compose path in `/deploy/podman` at the same time?
> 
> Then, I have second thoughts about putting it under `/deploy/quadlets` (wondering how many people know that name and whether the official name didn't change to "podman-systemd units" instead). So wanted to check what makes most sense: `quadlets`, `systemd`, `podman`?

I wanted to remove the podman compose folder, but I'm not sure if there is any current use (as I also see prometheus services, etc...). If that's clear, we can go ahead and remove it.

About naming, It's true that podman would be more obvious, but I don't want people to assume is a compose format, so I'm hesitant.

**Comment by fzdarsky** [comment]
> I wanted to remove the podman compose folder, but I'm not sure if there is any current use (as I also see prometheus services, etc...). If that's clear, we can go ahead and remove it.

Files in there are ancient (minus the prom ones), so pretty sure they no longe work. For the prom ones, might check with @vraiti and Yuri.

> About naming, It's true that podman would be more obvious, but I don't want people to assume is a compose format, so I'm hesitant.

As podman-compose is an extra tool that needs to be installed on top of podman, I believe that risk is small.

**Comment by oglok** [comment]
> > I wanted to remove the podman compose folder, but I'm not sure if there is any current use (as I also see prometheus services, etc...). If that's clear, we can go ahead and remove it.
> 
> Files in there are ancient (minus the prom ones), so pretty sure they no longe work. For the prom ones, might check with @vraiti and Yuri.
> 
> > About naming, It's true that podman would be more obvious, but I don't want people to assume is a compose format, so I'm hesitant.
> 
> As podman-compose is an extra tool that needs to be installed on top of podman, I believe that risk is small.

We keep introducing stuff with podman-compose. @avishayt has just added a new target to deploy Valkey. So, let's get this merged, and discuss in the next team meeting what to do with podman-compose deployments. A followup clean up would be very straight-forward.

**Comment by fzdarsky** [comment]
> We keep introducing stuff with podman-compose. @avishayt has just added a new target to deploy Valkey. So, let's get this merged, and discuss in the next team meeting what to do with podman-compose deployments. A followup clean up would be very straight-forward.

Valkey needs adding to the Quadlet manifests then and we should then remove the podman-compose asap, so that should there be more components they'll directly go into both Quadlet and Helm.

**Comment by oglok** [comment]
> > We keep introducing stuff with podman-compose. @avishayt has just added a new target to deploy Valkey. So, let's get this merged, and discuss in the next team meeting what to do with podman-compose deployments. A followup clean up would be very straight-forward.
> 
> Valkey needs adding to the Quadlet manifests then and we should then remove the podman-compose asap, so that should there be more components they'll directly go into both Quadlet and Helm.

I've added Valkey. There will need to be a follow up PR for cleaning up the podman-compose whenever it's clear.

**Review by fzdarsky — APPROVED** [review]

**Pull request merged**